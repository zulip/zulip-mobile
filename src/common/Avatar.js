/* @flow */
import React from 'react';

import ImageAvatar from './ImageAvatar';
import TextAvatar from './TextAvatar';
import { getFullUrl } from '../utils/url';

export default class Avatar extends React.PureComponent {

  props: {
    avatarUrl?: string,
    name: string,
    size: number,
    status?: string,
    realm?: string,
    shape?: 'square' | 'rounded' | 'circle',
    onPress?: () => void,
  };

  static defaultProps = {
    name: '',
    size: 32,
    realm: '',
    shape: 'rounded',
    onPress: () => {}
  };

  render() {
    const { avatarUrl, name, size, status, onPress, realm, shape } = this.props;

    return avatarUrl ?
      <ImageAvatar
        avatarUrl={getFullUrl(avatarUrl, realm)}
        size={size}
        status={status}
        onPress={onPress}
        shape={shape}
      /> :
      <TextAvatar
        name={name}
        size={size}
        status={status}
        onPress={onPress}
        shape={shape}
      />;
  }
}
