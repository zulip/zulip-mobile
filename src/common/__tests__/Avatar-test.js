import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../Avatar';

it('renders correctly', () => {
  const tree = renderer.create(
    <Avatar />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
