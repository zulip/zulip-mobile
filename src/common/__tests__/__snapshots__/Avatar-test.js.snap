exports[`test renders correctly 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "uri": undefined
    }
  }
  style={
    Object {
      "borderRadius": 16,
      "height": 32,
      "width": 32
    }
  } />
`;
